library/core/src/intrinsics.rs

```diff
@@ -2385,6 +2385,32 @@ extern "rust-intrinsic" {
     #[rustc_nounwind]
     pub fn raw_eq<T>(a: &T, b: &T) -> bool;
 
+    /// Lexicographically compare `[left, left + bytes)` and `[right, right + bytes)`
+    /// as unsigned bytes, returning negative if `left` is less, zero if all the
+    /// bytes match, or positive if `right` is greater.
+    ///
+    /// This underlies things like `<[u8]>::cmp`, and will usually lower to `memcmp`.
+    ///
+    /// # Safety
+    ///
+    /// * If `bytes` is zero, `left` and `right` must be non-null.
```

Comment:
```suggestion
    /// * Even if `bytes` is zero, `left` and `right` must be non-null.
```

otherwise this reads surprisingly if you don't already know about this